docs/command_line_reference.rst

```diff
@@ -674,11 +674,33 @@ Enable it with ``use_ssl:true``.
 **TLS/SSL Certificate Verification**
 
 Server certificate verification is controlled with the ``verify_certs`` boolean. The default value is `true`. To disable use ``verify_certs:false``.
-If ``verify_certs:true``, Rally will attempt to verify the certificate provided by Elasticsearch. If they are private certificates, you will also need to supply the private CA certificate using ``ca_certs:'/path/to/cacert.pem'``.
+If ``verify_certs:true``, Rally will attempt to verify the certificate provided by Elasticsearch. If they are private certificates, you will also need to supply the private CA certificate in PEM format using ``ca_certs:'/path/to/cacert.pem'``.
 
 You can also optionally present client certificates, e.g. if Elasticsearch has been configured with ``xpack.security.http.ssl.client_authentication: required`` (see also `Elasticsearch HTTP TLS/SSL settings <https://www.elastic.co/guide/en/elasticsearch/reference/current/security-settings.html#http-tls-ssl-settings>`_).
 Client certificates can be presented regardless of the ``verify_certs`` setting, but it's strongly recommended to always verify the server certificates.
 
+**CA & Client Certificate Format**
+
+Rally supports PEM format for CA and client certificates. Certificates in PKCS#12 formatted keystores will need to be exported to PEM format in order for Rally to use them. The ``openssl`` command can be used to export certificates from a PKCS#12 keystore. In the following example, PKCS#12 file ``elastic-certificates.p12`` was generated by the Elasticsearch ``elasticsearch-certutil`` command.
+
+* Export CA certificate:
+
+ ::
+
+   openssl pkcs12 -in elastic-certificates.p12 -cacerts -out cacert.pem
```

Comment:
Let's use the default `elasticsearch-certutil` file name?
```suggestion
   openssl pkcs12 -in elastic-stack-ca.p12 -cacerts -out cacert.pem
```

Reply:
`elastic-certificates.p12` is the default name of the single-instance bundle generated by `elasticsearch-certutil` and it always includes the CA certificate and not the CA private key. It was probably the wrong choice. The `-cacerts` option only applies to instance bundles generated by certutil, so pulling it from the CA bundle is a little simpler:

`openssl pkcs12 -in elastic-stack-ca.p12 -out cacert.pem`

I reason the CA bundle file name would be the most likely to remain the same and a user knowing their way around `openssl` can export the CA cert however they want. I will change it.

Reply:
I'm not sure I'm following. Where are you seeing `elastic-certificates.p12`? With the 7.16 and 8.0.0-beta1 Docker images, `elasticsearch-certutil ca` generates a `elastic-stack-ca.p12` file.

Also, have you tested that Rally can use those PEM files correctly?

Reply:
> I'm not sure I'm following. Where are you seeing `elastic-certificates.p12`? With the 7.16 and 8.0.0-beta1 Docker images, `elasticsearch-certutil ca` generates a `elastic-stack-ca.p12` file.

`elastic-certificates.p12` is the default file produced when executing the `cert` subcommand for `elasticsearch-certutil`. See the example at https://www.elastic.co/guide/en/elasticsearch/reference/current/certutil.html#_examples_5.

It was worded this way to encourage the use of a node certificate vs. the CA certificate for clients. And good point, I will definitely verify the exported certificates work with Rally correctly.

Please let me know if anything is unclear.

Reply:
I can confirm certificates exported using the `openssl` commands in this PR work correctly with Rally without further modification. The only consideration probably deserving further documentation is `verify_certs:true` since the hostname used in `--target-hosts` must match the one assigned to the node's instance certificate to complete server verification. In the default case for a single node, `elasticsearch-certutil` assigns the instance name "instance". 

```
openssl x509 -in client_cert.pem -noout -subject
subject= /CN=instance
```

All of this can get much more complex and I want to keep these docs as simple as possible to serve as a starting point for those looking to use server certificate verification and client authentication. That said, a better option and practice for a Rally cert/key would be to generate one specifically for benchmarking:

```sh
bin/elasticsearch-certutil cert --ca elastic-stack-ca.p12 --dns rally --pem --name rally
```

I look forward to updating the PR pending your feedback @pquentin.

Reply:
Sorry for the delay in reviewing, and thanks for iterating!

I agree that the idea is just to give instructions for the simplest use case, we don't have to document more complex cases. If documenting `--name rally` helps, then we can do it, but otherwise leave it like this.

While this now looks good, I have not managed to test this, because I can't make Elasticsearch work with p12 files. https://www.elastic.co/guide/en/elasticsearch/reference/current/configuring-tls-docker.html is actually using PEM (turns out there's a `--pem` flag!) and https://www.elastic.co/guide/en/elasticsearch/reference/7.16/security-basic-setup-https.html suggests using `elasticsearch-certutil http` instead, and requires more fiddling than I'd like.

What steps have you followed?